j2lint/cli.py

```diff
@@ -78,13 +77,33 @@ def sort_issues(issues):
     issues.sort(
         key=lambda issue: (
             issue.filename,
-            issue.linenumber,
+            issue.line_number,
             issue.rule.id
         )
     )
     return issues
 
 
+def get_linting_issues(file_or_dir_names, options, collection, checked_files):
+    """ checking errors and warnings in files """
+    lint_errors = {}
+    lint_warnings = {}
+    files = get_files(file_or_dir_names)
+
+    # Get linting issues
+    for file_name in files:
+        runner = Runner(collection, file_name, checked_files)
+        if file_name not in lint_errors:
+            lint_errors[file_name] = []
+        if file_name not in lint_warnings:
+            lint_warnings[file_name] = []
+        j2_errors, j2_warnings = runner.run()
+        lint_errors[file_name].extend(j2_errors)
+        lint_warnings[file_name].extend(j2_warnings)
+    print_linting_result(options, lint_warnings, lint_errors)
```

Comment:
it gives warning for too-many-branches , thats why I have added it in separate function

Reply:
I am not saying this is not a good idea my point is that you refactor like this for the `run` function:

```
def run()
    [...]
    remove_temporary_file(stdin_filename)
    results = get_linting_result()
    print_liniting_result(results)
    [...]
```

This way you keep separated the `get` and the `print` and it will allow for easier refactoring later

This way you can even use the results from `get_linting_error` to return the same value as it is returning now (cf my next comment in this file)